cairo/ethereum/cancun/body.cairo

```diff
@@ -0,0 +1,207 @@
+
+from mpt.hash_diff import (
+    hash_state_storage_diff,
+    hash_state_account_diff,
+    hash_account_diff_segment,
+    hash_storage_diff_segment,
+)
+from mpt.types import (
+    NodeStore,
+    OptionalUnionInternalNodeExtended,
+    MappingBytes32Bytes32,
+    MappingBytes32Address,
+)
+from mpt.trie_diff import compute_diff_entrypoint
+from mpt.utils import sort_account_diff, sort_storage_diff
+
+func main{
+    output_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr,
+    ecdsa_ptr: SignatureBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    ec_op_ptr: EcOpBuiltin*,
+    keccak_ptr: felt*,
+    poseidon_ptr: PoseidonBuiltin*,
+    range_check96_ptr: felt*,
+    add_mod_ptr: ModBuiltin*,
+    mul_mod_ptr: ModBuiltin*,
+}(start_index: felt, len: felt) {
+    alloc_locals;
+
+    // Fill-in the program inputs through the hints.
+    local block: Block;
+    local state: State;
+    local transactions_trie: TrieBytesOptionalUnionBytesLegacyTransaction;
+    local receipts_trie: TrieBytesOptionalUnionBytesReceipt;
+    local block_logs: TupleLog;
+    local block_hashes: ListHash32;
+    local gas_available: Uint;
+    local chain_id: U64;
+    local base_fee_per_gas: Uint;
+    local blob_gas_used: Uint;
+    local excess_blob_gas: U64;
+    %{ body_inputs %}
+
+    // TODO: once init merged
+    // let (initial_args_commitment) = init_commitment(block, state, transactions_trie, receipts_trie, block_logs, block_hashes, gas_available, chain_id, base_fee_per_gas, excess_blob_gas);
+
+    with state, transactions_trie, receipts_trie {
+        let (blob_gas_used, gas_available, block_logs) = _apply_body_inner(
+            index=start_index,
+            len=len,
+            transactions=block.value.transactions,
+            gas_available=gas_available,
+            chain_id=chain_id,
+            base_fee_per_gas=base_fee_per_gas,
+            excess_blob_gas=excess_blob_gas,
+            block_logs=block_logs,
+            block_hashes=block_hashes,
+            coinbase=block.value.header.value.coinbase,
+            block_number=block.value.header.value.number,
+            block_gas_limit=block.value.header.value.gas_limit,
+            block_time=block.value.header.value.timestamp,
+            prev_randao=block.value.header.value.prev_randao,
+            blob_gas_used=blob_gas_used,
+        );
+    }
+
+    finalize_state{state=state}();
+
+    default_dict_finalize(
+        cast(transactions_trie.value._data.value.dict_ptr_start, DictAccess*),
+        cast(transactions_trie.value._data.value.dict_ptr, DictAccess*),
+        0,
+    );
+    default_dict_finalize(
+        cast(receipts_trie.value._data.value.dict_ptr_start, DictAccess*),
+        cast(receipts_trie.value._data.value.dict_ptr, DictAccess*),
+        0,
+    );
+    // let (post_args_commitment) = init_commitment(block, state, transactions_trie, receipts_trie, block_logs, block_hashes, gas_available, chain_id, base_fee_per_gas, excess_blob_gas);
```

Comment:
i dont think we need to commit to excess_blob_gas or base_fee_per_gas

Reply:
we do if they're the input of the next step

Reply:
they are not afaik